src/js/verkleFFIBindings/getTreeKey.ts

```diff
@@ -0,0 +1,115 @@
+import { Context } from '../../../dist/cjs/wasm/rust_verkle_wasm'
+
+// This is equal to 2n + 256n * 64n.
+//
+// It is a constant that is used in the `getTreeKeyHashJs` function.
+// See the `getTreeKeyHashJs` function for more details.
+const firstChunk = new Uint8Array([2, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+
+// Implements `get_tree_key` as specified here: https://notes.ethereum.org/@vbuterin/verkle_tree_eip#Header-values
+export function getTreeKey(
+  context: Context,
+  address: Uint8Array,
+  treeIndex: Uint8Array,
+  subIndex: number,
+): Uint8Array {
+  const keyHash = getTreeKeyHash(context, address, treeIndex)
+
+  // Replace the last byte with the subIndex
+  keyHash[keyHash.length - 1] = subIndex
+  return keyHash
+}
+
+// Computes the hash of the address and treeIndex for use in the `getTreeKey` function
+//
+// Note: Tree Index is interpreted as a little endian number.
+function getTreeKeyHash(
+  context: Context,
+  address: Uint8Array,
+  treeIndexLE: Uint8Array,
+): Uint8Array {
+  if (address.length !== 32) {
+    throw new Error('Address must be 32 bytes')
+  }
+  if (treeIndexLE.length !== 32) {
+    throw new Error('Tree index must be 32 bytes')
+  }
+
+  // Below is the function we want to implement in JS from the spec.
+  //
+  // It is called pedersen_hash, but it is not a pedersen hash.
+  //
+  // Throughout the codebase it is named get_tree_key_hash to avoid this confusion.
+  //
+  // def pedersen_hash(inp: bytes) -> bytes32:
+  //  assert len(inp) <= 255 * 16
+  //  # Interpret input as list of 128 bit (16 byte) integers
+  //  ext_input = inp + b"\0" * (255 * 16 - len(inp))
+  //  ints = [2 + 256 * len(inp)] + \
+  //  int.from_bytes(ext_input[16 * i:16 * (i + 1)]) for i in range(255)]
+  //  return compute_commitment_root(ints).serialize()
```

Comment:
Hi Kev,
some additional comments here: we also need `getTreeKeyHash()` for optimization to not have this called over and over again with the same parameters from `getTreeKey()`, I will just expose this method in a quick follow-up release (so note is purely informational, nothing to do on your side).

Generally I am not so sure about the relation of `Context.getTreeKey()` and this standalone `getTreeKey()` coming from this file (and both exposed on the API level), can you clarify?

All the best
Holge